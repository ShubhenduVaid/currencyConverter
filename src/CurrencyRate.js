import React, { Component } from "react";
import PropTypes from "prop-types";

const rateContainer = {
  display: "flex",
  justifyContent: "center",
  flexDirection: "column"
};

const rateStyle = {
  textAlign: "center",
  color: "#00b9ff",
  fontStyle: "italic",
  fontSize: "larger"
};

export default class CurrencyRate extends Component {
  handleFraction = value => {
    const fraction = value.split(".")[1];
    if (fraction && fraction.length > 2) {
      return (+value).toFixed(4);
    }
    return value;
  };

  render() {
    const { toCurrency, fromCurrency, factor } = this.props;
    let rate = "";
    if (toCurrency === fromCurrency) {
      rate = "1";
    } else if (toCurrency === "USD") {
      rate = 1 / factor["USD" + fromCurrency];
    } else if (fromCurrency !== "USD" && toCurrency !== "USD") {
      rate = (1 / factor["USD" + fromCurrency]) * factor["USD" + toCurrency];
    } else {
      rate = 1 * factor["USD" + toCurrency];
    }
    rate = this.handleFraction(rate.toString());
    const message = `1 ${fromCurrency} = ${rate} ${toCurrency}`;
    return (
      <div style={rateContainer}>
        <p style={rateStyle}>{message}</p>
      </div>
    );
  }
}

CurrencyRate.propTypes = {
  toCurrency: PropTypes.string,
  fromCurrency: PropTypes.string,
  factor: PropTypes.object
};
