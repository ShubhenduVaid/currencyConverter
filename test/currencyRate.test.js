import React from "react";
import renderer from "react-test-renderer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import CurrencyRate from "../src/currencyRate";

configure({ adapter: new Adapter() });

describe("Check CurrencyRate component snapshots", () => {
  it("CurrencyRate component snapshot", () => {
    const component = renderer.create(<CurrencyRate />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
