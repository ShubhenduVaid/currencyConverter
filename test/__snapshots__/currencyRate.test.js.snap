// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check CurrencyRate component snapshots CurrencyRate component snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <p
    style={
      Object {
        "color": "#00b9ff",
        "fontSize": "larger",
        "fontStyle": "italic",
        "textAlign": "center",
      }
    }
  >
    1 undefined = 1 undefined
  </p>
</div>
`;
